src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -405,6 +405,20 @@ PeerImp::json()
     return ret;
 }
 
+bool
+PeerImp::supportsProtocolVersion(const char* target) const
+{
+    auto const goal = parseProtocolVersions(target);
+    // The values are passed in as hard-coded strings. Make sure that
+    // string represents a single, valid supported version.
+    assert(goal.size() == 1 &&
+        negotiateProtocolVersion(goal));
+
+    return goal.size() == 1 &&
+        protocol_ >= goal.front();
```

Comment:
I'm not quite sure I like this.

First, just because feature X is supported with `XRPL/2.1` doesn't mean it's supported on version `XRPL/2.2` or later. Second, the API is messy. It requires me (as the prorammer) to keep mental track of which features are supported with each version. This can and will quickly become unwieldy.

I do believe this functionality is good to have, but I propose a different way of accessing it:

```
enum class ProtocolFeature
{
    ValidatorListPropagation = 1,
    SomethingElse = 2,
};

bool PeerImp::supportsFeature(ProtocolFeature f)
{
    switch (f)
    {
    case ProtocolFeature::ValidatorListPropagation: 
        return protocol_ >= make_protocol(2, 1);
    case ProtocolFeature::SomethingElse:
        return protocol >= make_protocol(2, 0) && protocol < make_protocol(3,1);
    }
    return false;
}
```

If we felt like it (and I don't think we need to go the extra mile) with a little more work and the help of `constexpr` we could do a little better, and reduce this to an array index and/or map lookup.

Reply:
I like this a lot better, too. I'll make the change.